### PR TITLE
Update DefaultAPISidebar and GroupData for new events structure

### PR DESCRIPTION
**Pull request by wbamberg**
Here's a PR to update GroupData for the new events organization.

It's technically part of https://github.com/mdn/sprints/issues/1315, although https://github.com/mdn/sprints/issues/1314 is where all the action is.

Before it can be merged:

1. we need tests for DefaultAPISidebar (https://github.com/mdn/kumascript/pull/1135 needs to be merged)
2. we need to update DefaultAPISidebar to handle this new structure (PR not yet filed)

But since this is quite a complicated change I'd love to get some review of it now anyway.

The basic idea is outlined in https://github.com/mdn/sprints/issues/1314#issuecomment-487251021 with an extra twist at https://github.com/mdn/sprints/issues/1314#issuecomment-487298961.

So to implement it I went through every group that had some events in the `events` property, and:

* if the API was archived (or was otherwise useless), deleted the group
* deleted any events that were undocumented (since we have nothing to link to, there)
* deleted any events that belonged to interfaces that belonged to the interface (e.g. see the `FileReader` example in https://github.com/mdn/sprints/issues/1314#issuecomment-487251021 )
* for any events that belong to interfaces outside the group (e.g. see the "Clipboard API" example in https://github.com/mdn/sprints/issues/1314#issuecomment-487251021 ), kept them and prepended "Interfacename: ".

Where an event targeted multiple interfaces:
* if this was just bubbling, I included only the original target
* if it wasn't just bubbling and they were distinct events, I listed both

I used this: https://docs.google.com/spreadsheets/d/1-G5q6sqiT6iSz-O3tJMhdDjBWazXUvXeVoyP6eWH4X0/edit#gid=0 to track the work. It might help the review.

### CODE REVIEW

File: tests/macros/GroupData.test.js [hunk 1]
```diff
@@ -14,7 +14,7 @@ const permittedCharacters = {
     interface: /^[A-Z][\w.]+$/,
     property: /^[\w.]+$/,
     method: /^[\w.()]+$/,
-    event: /^[\w()]+$/,
+    event: /^[\w]+:\ [\w]+$/,
```
Comment (ExE-Boss) [inline tests/macros/GroupData.test.js]: The space doesn't need to be escaped:
```suggestion
    event: /^[\w]+: [\w]+$/,
```

**Comment by wbamberg** [comment]
I did laboriously sort one of them, but then thought we should have the macro sort these on render, as for BCD. But we should also have the JSON sorted, and have this tested, just to make contribution easier. But I'd prefer to do that in a follow-up I think, to keep this PR relatively simple.

**Comment by wbamberg** [comment]
Tests for DefaultAPISidebar (https://github.com/mdn/kumascript/pull/1135) landed, so I've updated this PR with changes to DefaultAPISidebar to handle the new event structure, along with updates to the tests.

I was tempted to update the string concatenation code to use template literals, but decided to keep changes minimal in this PR.

@Elchi3 , would you mind taking another look? The new commit doesn't touch GroupData itself, only DefaultAPISidebar and its tests.

Apart from running the tests you could see what it looks like by scraping something like:

    docker exec kuma_worker_1 python ./manage.py scrape_document https://developer.mozilla.org/en-US/docs/Web/API/Pointer_events --depth 10

or 

    docker exec kuma_worker_1 python ./manage.py scrape_document https://developer.mozilla.org/en-US/docs/Web/API/Touch_events --depth 10

into a local Kuma.

File: macros/DefaultAPISidebar.ejs [hunk 2]
```diff
@@ -83,28 +82,32 @@ function buildSublist(pages, title) {
 function buildIFList(interfaces, title) {
   var result = '<li class="toggle"><details open><summary>' + title + '</summary><ol>';
 
-  for (var i = 0; i < interfaces.length; i++) {
-    var url = interfaces[i].replace('()', '').replace('.', '/');
-    result += '<li><a href="' + APIHref + '/' + url + '"><code>' + interfaces[i] + '</code></a></li>';
+  for (let interface of interfaces) {
+    result += '<li><a href="' + APIHref + '/' + interface.target + '"><code>' + interface.text + '</code></a></li>';
```
Comment (Elchi3) [inline macros/DefaultAPISidebar.ejs]:
```suggestion
    result += '<li><a href="' + APIHref + '/' + interface.target + '">' + interface.text + '</a></li>';
```

File: macros/DefaultAPISidebar.ejs [hunk 3]
```diff
@@ -83,28 +82,32 @@ function buildSublist(pages, title) {
...
-  for (var i = 0; i < events.length; i++) {
-    result += '<li><a href="' + EventHref + '/' + events[i] + '"><code>' + events[i] + '</code></a></li>';
-  }
-
-  result += '</ol></details></li>';
-
-  return result;
+function convert(itemName) {
+    return {
+      text: itemName,
```
Comment (Elchi3) [inline macros/DefaultAPISidebar.ejs]:
```suggestion
      text: `<code>${itemName}</code>`,
```

File: macros/DefaultAPISidebar.ejs [hunk 4]
```diff
@@ -83,28 +82,32 @@ function buildSublist(pages, title) {
...
+    }
 }
 
+function convertEvent(eventName) {
+    const eventComponents = eventName.split(': ');
+    if (eventComponents.length !== 2) {
+      return null;
+    }
+    return {
+      text: eventName,
```
Comment (Elchi3) [inline macros/DefaultAPISidebar.ejs]:
```suggestion
      text: `<code>${eventComponents[0]}</code>: <code>${eventComponents[1]}</code>`,
```

**Comment by wbamberg** [comment]
Thanks for the review @Elchi3 and for your suggestion, it makes sense. I guess we should do the same for APIRef too.

I'v added and tested your suggestion. I also reinstated the data for the Battery API in the light of https://discourse.mozilla.org/t/proposal-clean-groupdata/39610/3, since this API does have docs.

**Pull request merged**